joomla/joomla-cms
# [5.1] Proper support for avif/webp images

Pull Request for Issue # .

### Summary of Changes

- add missing avif support in the `Image` class
- The resize plugin cannot create an image from string for AVIF files (someone needs to open an issue to php-src for this)
- both webp and avif cannot be supported for editing because browsers are inconsistent atm on the `canvas.toDataURL()`
spec: [toDataURL](https://html.spec.whatwg.org/multipage/canvas.html#dom-canvas-todataurl)


### Testing Instructions

- Either run `npm I` or use the package from the PR
- extract and upload the files [Archive.zip](https://github.com/joomla/joomla-cms/files/15030023/Archive.zip)
- Delete/rename the files

- Delete the files. Edit the Media Action plugin resize with some values (ie 800 and 500)  and check that the images on upload are getting resized.

### Actual result BEFORE applying this Pull Request



### Expected result AFTER applying this Pull Request



### Link to documentations
Please select:
